src/http/one/Tokenizer.h

```diff
@@ -9,68 +9,52 @@
 #ifndef SQUID_SRC_HTTP_ONE_TOKENIZER_H
 #define SQUID_SRC_HTTP_ONE_TOKENIZER_H
 
-#include "parser/Tokenizer.h"
+#include "parser/forward.h"
+
+class SBuf;
 
 namespace Http {
 namespace One {
 
 /**
- * Lexical processor extended to tokenize HTTP/1.x syntax.
+ * Extracts either an HTTP/1 token or a complete HTTP/1
+ * quoted-string (and sets the quoted accordingly).
+ * Unescapes escaped characters in HTTP/1.1 quoted strings.
+ *
+ * Governed by:
+ *  - RFC 1945 section 2.1
+ *  "
+ *    A string of text is parsed as a single word if it is quoted using
+ *    double-quote marks.
+ *
+ *        quoted-string  = ( <"> *(qdtext) <"> )
+ *
+ *        qdtext         = <any CHAR except <"> and CTLs,
+ *                         but including LWS>
+ *
+ *    Single-character quoting using the backslash ("\") character is not
+ *    permitted in HTTP/1.0.
+ *  "
+ *
+ *  - RFC 7230 section 3.2.6
+ *  "
+ *    A string of text is parsed as a single value if it is quoted using
+ *    double-quote marks.
  *
- * \see ::Parser::Tokenizer for more detail
+ *    quoted-string  = DQUOTE *( qdtext / quoted-pair ) DQUOTE
+ *    qdtext         = HTAB / SP /%x21 / %x23-5B / %x5D-7E / obs-text
+ *    obs-text       = %x80-FF
+ *  "
+ *
+ * \param http1p0 HTTP/1.0 does not permit \-escaped characters
+ * \param tokenPrefixResult function return value when input is a token prefix
+ * \returns tokenPrefixResult if input contains nothing but a token (prefix)
+ * \returns true (and sets the value) if input starts with a token or quoted-string
+ * \returns false (and leaves the value intact) if input does not start with a token or quoted-string
+ * \throws on syntax violations
```

Comment:
Doxygen syntax requires naming the exception type which is thrown.
https://www.stack.nl/~dimitri/doxygen/manual/commands.html#cmdthrow

Reply:
AFAICT, Doxygen does not require naming the exception type -- doing so would actually promote bad code design. Doxygen does require naming the exception object when describing what is being thrown. In those rare cases where we should use `\throw`, starting with something like `\throws exception ...` may work well enough.

FWIW, in most cases, we should _not_ document that a function throwing at all because any function can throw by default. This function can be considered a special case, but I am not against removal of that `\throws` line if that is what other prefer.

Reply:
To avoid confusion, I removed this `\throws` line from there (3cdffa). BTW, this is in line with other Squid code documentation, which does not use this `\throws`. I found only 1-2 uses in all Squid sources.